### PR TITLE
board: arm: mec15modular_assy6885: Enable RTOS, PS2, PWM and ADC drivers

### PR DESCRIPTION
**Pull request by albertofloyd**
Enable RTOS, PS2, PWM and ADC for modular MEC15xx too.
Add extra step to build flashable image.

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

### CODE REVIEW

File: boards/arm/mec1501modular_assy6885/CMakeLists.txt [hunk 1]
```diff
@@ -6,3 +6,48 @@
 
 zephyr_library()
 zephyr_library_sources(pinmux.c)
+
+if(DEFINED ENV{EVERGLADES_SPI_GEN})
+  # Grab it from environment variable if defined
+  set(EVERGLADES_SPI_GEN $ENV{EVERGLADES_SPI_GEN})
```
Comment (SebastianBoe) [inline boards/arm/mec1501modular_assy6885/CMakeLists.txt]: This needs documentation.

**Review by SebastianBoe — CHANGES_REQUESTED** [review]

File: boards/arm/mec1501modular_assy6885/CMakeLists.txt (hunk 1 above)
Comment (franciscomunoz) [reply boards/arm/mec1501modular_assy6885/CMakeLists.txt]: @SebastianBoe would it be enough to mention in the documentation that we should export EVERGLADES_SPI_GEN?

Comment (SebastianBoe) [reply boards/arm/mec1501modular_assy6885/CMakeLists.txt]: yes.

Comment (albertofloyd) [reply boards/arm/mec1501modular_assy6885/CMakeLists.txt]: Added more info in comment here and reference to board documentation for futher details.
SPI generation tool documentation is added as part board doc update PR https://github.com/zephyrproject-rtos/zephyr/pull/19731/

**Review by SebastianBoe — APPROVED** [review]

**Review by franciscomunoz — APPROVED** [review]

**Pull request merged**